packages/domain-search/src/components/DomainsFullCart/DomainsFullCart.tsx

```diff
@@ -1,26 +1,106 @@
+import {
+	// eslint-disable-next-line wpcalypso/no-unsafe-wp-apis
+	__unstableMotion as motion,
+	Card,
+	CardHeader,
+	__experimentalHeading as Heading,
+	CardBody,
+	Button,
+	CardFooter,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useViewportMatch } from '@wordpress/compose';
+import { close } from '@wordpress/icons';
+import { useI18n } from '@wordpress/react-i18n';
+import { useCallback, useRef, useEffect } from 'react';
 import { useDomainSearch } from '../DomainSearch/DomainSearch';
 import { DomainsFullCartItems } from './Items';
+import { DomainsFullCartSummary } from './Summary';
+
+import './style.scss';
 
 const DomainsFullCart = ( { children }: { children?: React.ReactNode } ) => {
-	const { isFullCartOpen } = useDomainSearch();
+	const { isFullCartOpen, closeFullCart, onContinue } = useDomainSearch();
+	const { __ } = useI18n();
+	const isMobile = ! useViewportMatch( 'small' );
+
+	const fullCartRef = useRef< HTMLDivElement >( null );
+
+	const handleClickOutside = useCallback(
+		( event: MouseEvent ) => {
+			if ( fullCartRef.current && ! fullCartRef.current.contains( event.target as Node ) ) {
+				closeFullCart();
+			}
+		},
+		[ closeFullCart ]
+	);
 
-	if ( ! isFullCartOpen ) {
-		return null;
-	}
+	useEffect( () => {
+		if ( isFullCartOpen ) {
+			document.addEventListener( 'mousedown', handleClickOutside );
+			return () => {
+				document.removeEventListener( 'mousedown', handleClickOutside );
+			};
+		}
+	}, [ isFullCartOpen, handleClickOutside ] );
```

Comment:
I'm thinking of extracting this into a hook and moving it to a different, more generic package because it has zero to do with the full cart itself. I also wonder if Gutenberg exposes this functionality 🤔 Haven't found anything.